src/evaluate-conditions.js

```diff
@@ -93,6 +90,7 @@ export default function evaluate (model, value, getPreviousValue) {
 
   const getValue = pathFinder(value, getPreviousValue)
 
+  retModel.properties = _.clone(model.properties)
```

Comment:
`_.cloneDeep` was too expensive and shallow clone was all that was needed since this was a recursive function. I was able to verify this work with the *heterogenous* array test where it checks the equality of the possible schemas it produces on the recursive call to `evaluate`. That operation required that the object it was testing be separate objects.